elements/pf-progress/pf-progress.ts

```diff
@@ -0,0 +1,189 @@
+ *          {@default `#3e8635`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-warning__bar--BackgroundColor
+ *          Background color of the progress bar when variant is warning.
+ *          {@default `#f0ab00`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-danger__bar--BackgroundColor
+ *          Background color of the progress bar when variant is danger.
+ *          {@default `#c9190b`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-success__status-icon--Color
+ *          Color of the status icon when variant is success.
+ *          {@default `#3e8635`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-warning__status-icon--Color
+ *          Color of the status icon when variant is warning.
+ *          {@default `#f0ab00`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-danger__status-icon--Color
+ *          Color of the status icon when variant is danger.
+ *          {@default `#c9190b`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-success--m-inside__measure--Color
+ *          Color of the progress bar measure when variant is success and measure location is inside.
+ *          {@default `#ffffff`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-outside__measure--FontSize
+ *          Font size of the progress bar measure when measure location is outside.
+ *          {@default `0.875rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-sm__bar--Height
+ *          Height of the progress bar when the size is small.
+ *          {@default `0.5rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-sm__description--FontSize
+ *          Font size of the progress bar description when the size is small.
+ *          {@default `0.875rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-lg__bar--Height
+ *          Height of the progress bar when the size is large.
+ *          {@default `1.5rem`}
+ *
+ */
+@customElement('pf-progress')
+export class PfProgress extends LitElement {
+  static readonly styles = [styles];
+
+  static readonly formAssociated = true;
+
+  /** Represents the value of the progress bar */
+  @property({ reflect: true, type: Number })
+    value = 0;
+
+  /** Title above the progress bar */
+  @property()
+    title = '';
+
+  /** Maximum value for the progress bar */
+  @property({ type: Number, reflect: true })
+    max = 100;
+
+  /** Minimum value for the progress bar */
+  @property({ type: Number, reflect: true })
+    min = 0;
+
+  /** Size of the progress bar (height) */
+  @property()
+    size: 'sm' | 'lg' | '' = '';
+
+  /** Where the percentage will be displayed with the progress element */
+  @property({ attribute: 'measure-location' })
+    measureLocation: '' | 'outside' | 'inside' | 'none' = '';
+
+  /** Variant of the progress bar */
+  @property()
+    variant: '' | 'success' | 'danger' | 'warning' = '';
+
+  get #calculatedPercentage(): number {
+    const { value, min, max } = this;
+    const percentage = Math.round((value - min) / (max - min) * 100);
+    if (isNaN(percentage) || percentage < 0) {
```

Comment:
```suggestion
    if (Number.isNaN(percentage) || percentage < 0) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN